python/docker-compose.yml

```diff
@@ -11,5 +11,6 @@ services:
       - ./logs:/home/ml4ir/python/logs
       - ./data:/home/ml4ir/python/data
       - ./models:/home/ml4ir/python/models
+      - ./ml4ir/:/home/ml4ir/python/ml4ir
```

Comment:
@ducouloa So how does this work if `/home/ml4ir/python/ml4ir` already exists?

Reply:
This is still dangerous IMO, because you might think there is something in the docker file, test it and ship it, and then realize it was only a local change/you forgot to rebuild the image/etc.

Why not just mount the configs? I don't understand the pushback.

Reply:
I'm not sure I'm push-backing anything here?

My reasoning behind mounting the whole ml4ir dir is exactly to be able to test all the local change made, not only the config change, otherwise you need to `docker-compose build` each time you want to do an integration test with local change on the python code. And obviously, missing a build step while testing is a trap we want to avoid in the future right?

About shipping local change this is a great point until we kept our delivery process beeing manual on a local machine of a designed volunteer, which definitely sounds like a bad practice. I'd rather fix this delivery process IMO

Reply:
> otherwise you need to docker-compose build each time you want to do an integration test with local change on the python code

This is the right way to do it IMO. The code needs to be frozen. You can alter the configs all you want. Mounting the code means there is a high change the docker image and the code which was just integration tested might not be the same. If this change only affects local, I guess it is okay. But think of docker being a way to ship `ml4ir` to end users. Mounting code just messes it up.

tl;dr -> rebuilding each time should be the right thing to do.

Reply:
You seem strongly opinionated about this change, so I'll made the modification as you wish.

Note that the [official Docker documentation](https://docs.docker.com/storage/bind-mounts/#mount-into-a-non-empty-directory-on-the-container) reference this practice as "can be beneficial:

> If you bind-mount into a non-empty directory on the container, the directory’s existing contents are obscured by the bind mount. **This can be beneficial**, such as **when you want to test a new version of your application without building a new image**